File: www/cordova-grpc.js

var exec = require('cordova/exec');

var PLUGIN_NAME = 'CordovaGrpc';

var CordovaGrpc = {
  startServer: function(port, cb) {
    exec(cb, null, PLUGIN_NAME, 'startServer', [port]);
  },
  stopServer: function(cb) {
    exec(cb, null, PLUGIN_NAME, 'stopServer', []);
  },
  getPort: function(cb) {
    exec(cb, null, PLUGIN_NAME, 'getPort', []);
  }
};

module.exports = CordovaGrpc;

File: src/ios/CordovaGrpc.m
#import "CordovaGrpc.h"

#import <Cordova/CDVAvailability.h>

@implementation CordovaGrpc

- (void)pluginInitialize {
}

-(void)startServer:(CDVInvokedUrlCommand *)command {
  NSInteger port = [command.arguments objectAtIndex:0];
  NSLog(@"%@", port);
  CDVPluginResult* pluginResult = [CDVPluginResult resultWithStatus:CDVCommandStatus_OK];
  [self.commandDelegate sendPluginResult:pluginResult callbackId:command.callbackId];
}

-(void)stopServer:(CDVInvokedUrlCommand *)command {

}

-(void)getPort:(CDVInvokedUrlCommand *)command {

}

@end
